pkg/commands/service.go

```diff
@@ -13,27 +13,26 @@ import (
 // Service : A docker Service
 type Service struct {
 	Name          string
-	ID            string
 	OSCommand     *OSCommand
 	Log           *logrus.Entry
-	Container     *Container
+	Containers    []*Container
 	DockerCommand LimitedDockerCommand
 }
 
 // GetDisplayStrings returns the dispaly string of Container
 func (s *Service) GetDisplayStrings(isFocused bool) []string {
 
-	if s.Container == nil {
+	if len(s.Containers) == 0 {
 		return []string{utils.ColoredString("none", color.FgBlue), s.Name, ""}
 	}
 
-	cont := s.Container
+	cont := s.Containers[0]
```

Comment:
given how often we're using this `s.Container[0]` approach, could we factor that out into a function like
```
func (s *Service) SelectedContainer() Container {
	return s.Containers[0]
}
```

That way, when we add the code for dealing with multiple containers, we only need to change it in one place

Reply:
We certainly can, though I'm not sure how useful that would be overall.  `commands.Service` doesn't really have `gui` responsibilities, so delegating "selection" responsibilities to it seems to conflate concerns a bit.  A function that *did* have `gui` responsibilities might look like

``` golang
func (gui *Gui) selectedContainerForService(s *Service) *Container {
	idx := gui.State.Panels.Services.SelectedContainer[s.ID]
	return s.Containers[idx]
}
```

I'm not sure how relevant I would consider this for dealing with multiple containers, though.  The majority of the `s.Containers[0]` cases are treating the Service as a proxy for the Container, not acting on a "selection", and when we have the ability to select Containers for a Service that behavior doesn't make sense – we should be able to treat that selection exactly like any other Container.  At that point, Services need to stop being proxies for Containers, and start behaving like groups of Containers (e.g. "Logs" show the interleaved logs from all containers for the Service, "Container Config" goes away, "Stats" either presents an aggregate view or goes away, etc.).

A reasonable case could be made that when only a single Container exists for a Service, the two should be treated interchangeably.  In my own explorations around that, I've found having a clear delineation between Service and Container in the UI helps both predictability and implementation, and provides a transparent solution to other issues for free (e.g. the implicit vs explicit container name issue goes away if you're not collapsing Services and Containers into the same item).

Reply:
yeah I agree. If I recall correctly, I did use `docker-compose logs <service name>` being used at some point but decided it was a bit jarring to see the service name at the side when it was obvious from the context what service you were looking at, so I just showed the container logs instead.

I agree about having a clear separation between the two, and I'm happy for that to be bigger discussion down the road. At any rate, feel free to use that `selectedContainerForService` function if you wish, I'll defer to your judgement.

Reply:
Absolutely – in the current UI, with the current implicit tie between a Service and a single Container, it would absolutely feel bad to see the container name next to every log line.  Container logs make perfect sense in that context; the issue comes from the fact that we're going to need to leave that context behind.

For now, I'm content to leave this as `service.Containers[0]` since it's both temporary (🤞) and the simplest translation of the existing logic.